crates/transport/Cargo.toml

```diff
@@ -27,7 +27,10 @@ tower.workspace = true
 url.workspace = true
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]
-wasm-bindgen-futures = "0.4"
+wasm-bindgen-futures = { version = "0.4", optional = true }
+
+[target.'cfg(target_arch = "wasm32")'.features]
```

Comment:
i am relatively sure that target-specific features don't work, per an [old old old issue](https://github.com/rust-lang/cargo/issues/1197). 

so in this case we would omit the `.features` block and rely on the user to specify separately I guess? that kinda sucks but 🤔

Reply:
just moving to features should be fine as the dependency will only get compiled if the cfg is enabled